File: app/src/main/java/org/d3if3068/assesment001/MyDatabaseHelper.java
package org.d3if3068.assesment001;

import android.content.ContentValues;
import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.widget.Toast;

import androidx.annotation.Nullable;

class MyDatabaseHelper extends SQLiteOpenHelper {

    private Context context;
    private final static String DATABASE_NAME = "HistoryDatabase.db";
    private final static int DATABASE_VERSION = 1;

    private static final String TABLE_NAME = "history";
    private static final String COLUMN_ID = "_id";
    private static final String COLUMN_CELCIUS = "celcius";
    private static final String COLUMN_FAHRENHEIT = "fahrenheit";

    public MyDatabaseHelper(@Nullable Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
        this.context = context;
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        String query = "CREATE TABLE " + TABLE_NAME +
                " (" + COLUMN_ID + " INTEGER PRIMARY KEY AUTOINCREMENT, " +
                COLUMN_CELCIUS + " TEXT, " +
                COLUMN_FAHRENHEIT + " TEXT);";
        db.execSQL(query);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db,int i,int i1) {
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_NAME);
        onCreate(db);
    }

    void addHistory(String celcius, String fahrenheit){
        SQLiteDatabase db = this.getWritableDatabase();
        ContentValues cv = new ContentValues();

        cv.put(COLUMN_CELCIUS, celcius);
        cv.put(COLUMN_FAHRENHEIT, fahrenheit);

        long result = db.insert(TABLE_NAME, null, cv);

        if(result == -1){
            Toast.makeText(context, "Failed", Toast.LENGTH_SHORT).show();
        }else {
            Toast.makeText(context, "Success", Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/org/d3if3068/assesment001/MyDatabaseHelper.kt
package org.d3if3068.assesment001

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.widget.Toast

class MyDatabaseHelper(private val context: Context?) : SQLiteOpenHelper(
    context, DATABASE_NAME, null, DATABASE_VERSION
) {
    override fun onCreate(db: SQLiteDatabase) {
        val query = "CREATE TABLE " + TABLE_NAME +
                " (" + COLUMN_ID + " INTEGER PRIMARY KEY AUTOINCREMENT, " +
                COLUMN_CELCIUS + " TEXT, " +
                COLUMN_FAHRENHEIT + " TEXT);"
        db.execSQL(query)
    }

    override fun onUpgrade(db: SQLiteDatabase, i: Int, i1: Int) {
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_NAME)
        onCreate(db)
    }

    fun addHistory(celcius: String?, fahrenheit: String?) {
        val db = this.writableDatabase
        val cv = ContentValues()
        cv.put(COLUMN_CELCIUS, celcius)
        cv.put(COLUMN_FAHRENHEIT, fahrenheit)
        val result = db.insert(TABLE_NAME, null, cv)
        if (result == -1L) {
            Toast.makeText(context, "Failed", Toast.LENGTH_SHORT).show()
        } else {
            Toast.makeText(context, "Success", Toast.LENGTH_SHORT).show()
        }
    }

    fun readAllData(): Cursor? {
        val query = "SELECT * FROM " + TABLE_NAME
        val db = this.readableDatabase
        var cursor: Cursor? = null
        if (db != null) {
            cursor = db.rawQuery(query, null)
        }
        return cursor
    }

    fun deleteDataById(id: String): Int {
        val db = this.writableDatabase
        return db.delete(TABLE_NAME, COLUMN_ID + "=?", arrayOf(id))
    }

    companion object {
        private const val DATABASE_NAME = "HistoryDatabase.db"
        private const val DATABASE_VERSION = 1
        private const val TABLE_NAME = "history"
        private const val COLUMN_ID = "_id"
        private const val COLUMN_CELCIUS = "celcius"
        private const val COLUMN_FAHRENHEIT = "fahrenheit"
    }
}